sky/jobs/recovery_strategy.py

```diff
@@ -368,6 +380,13 @@ def _launch(self,
                         f'{gap_seconds:.1f} seconds.')
             time.sleep(gap_seconds)
 
+    def trigger_retry_on_failure(self) -> bool:
+        """Trigger a retry on failure."""
```

Comment:
The name of this method is a bit misleading since this does not "trigger" a retry, rather just records it. Maybe rename to `log_retry_on_failure` or `record_retry_on_failure`?

Docstr could also be updated to something like:
`"""Records a retry event after a job failure and returns if more retries should be attempted."""`

Reply:
Good point! I renamed it to `should_restart_on_failure` with the docstr updated. Wdyt?